rasa/core/domain.py

```diff
@@ -216,18 +220,18 @@ def from_path(cls, path: Text, skill_imports: SkillSelector) -> "Domain":
         return domain
 
     @classmethod
-    def from_file(cls, path: Text) -> "Domain":
-        return cls.from_yaml(rasa.utils.io.read_file(path))
+    def from_file(cls, path: Text, log_warning: bool = True) -> "Domain":
+        return cls.from_yaml(rasa.utils.io.read_file(path), log_warning)
 
     @classmethod
-    def from_yaml(cls, yaml: Text) -> "Domain":
+    def from_yaml(cls, yaml: Text, log_warning: bool = True) -> "Domain":
         cls.validate_domain_yaml(yaml)
         data = rasa.utils.io.read_yaml(yaml)
-        return cls.from_dict(data)
+        return cls.from_dict(data, log_warning)
 
     @classmethod
-    def from_dict(cls, data: Dict) -> "Domain":
-        utter_templates = cls.collect_templates(data.get("templates", {}))
+    def from_dict(cls, data: Dict, log_warning: bool = True) -> "Domain":
```

Comment:
I feel like there's a better way to handle this, but maybe if we don't also call in the CLI we don't have to worry about this, see above comments. I think it makes sense to validate once on training (e.g. agent initialization) and then once on loading, if possible